front/src/common/api/osrdEditoastApi.ts

```diff
@@ -42,24 +42,115 @@ const osrdEditoastApi = generatedEditoastApi.enhanceEndpoints({
+    },
+
+    // Studies handling
+    getProjectsByProjectIdStudies: {
+      providesTags: (result) => [
+        'studies',
+        { type: 'studies', id: 'LIST' },
+        ...(result?.results || []).map(({ id }) => ({
+          type: 'studies' as const,
+          id,
+        })),
+      ],
+    },
+    getProjectsByProjectIdStudiesAndStudyId: {
+      providesTags: (_result, _error, args) => ['studies', { type: 'studies', id: args.studyId }],
+    },
     postProjectsByProjectIdStudies: {
-      invalidatesTags: ['studies', 'projects'],
+      invalidatesTags: (_result, _error, args) => [
+        { type: 'projects', id: args.projectId },
+        { type: 'studies', id: 'LIST' },
+      ],
     },
     patchProjectsByProjectIdStudiesAndStudyId: {
-      invalidatesTags: ['studies', 'projects'],
+      invalidatesTags: (_result, _error, args) => [
+        { type: 'projects', id: args.projectId },
+        { type: 'studies', id: 'LIST' },
+        { type: 'studies', id: args.studyId },
+      ],
     },
     deleteProjectsByProjectIdStudiesAndStudyId: {
-      invalidatesTags: ['studies', 'projects'],
+      invalidatesTags: (_result, _error, args) => [
+        { type: 'projects', id: args.projectId },
+        { type: 'studies', id: 'LIST' },
+      ],
+    },
+
+    // Scenari handling
+    getProjectsByProjectIdStudiesAndStudyIdScenarios: {
+      providesTags: (result) => [
+        'scenarios',
+        { type: 'scenarios', id: 'LIST' },
+        ...(result?.results || []).map(({ id }) => ({
+          type: 'scenarios' as const,
+          id,
+        })),
+      ],
+    },
+    getProjectsByProjectIdStudiesAndStudyIdScenariosScenarioId: {
+      providesTags: (_result, _error, args) => [
+        'scenarios',
+        { type: 'scenarios', id: args.scenarioId },
+      ],
     },
     postProjectsByProjectIdStudiesAndStudyIdScenarios: {
-      invalidatesTags: ['scenarios', 'studies', 'projects'],
+      invalidatesTags: (_result, _error, args) => [
+        { type: 'projects', id: 'LIST' },
+        { type: 'projects', id: args.projectId },
+        { type: 'studies', id: 'LIST' },
+        { type: 'studies', id: args.studyId },
+        { type: 'scenarios', id: 'LIST' },
+      ],
     },
     patchProjectsByProjectIdStudiesAndStudyIdScenariosScenarioId: {
-      invalidatesTags: ['scenarios', 'studies', 'projects'],
+      invalidatesTags: (_result, _error, args) => [
+        { type: 'projects', id: 'LIST' },
+        { type: 'projects', id: args.projectId },
+        { type: 'studies', id: 'LIST' },
+        { type: 'studies', id: args.studyId },
+        { type: 'scenarios', id: 'LIST' },
+        { type: 'scenarios', id: args.scenarioId },
+      ],
     },
     deleteProjectsByProjectIdStudiesAndStudyIdScenariosScenarioId: {
-      invalidatesTags: ['scenarios', 'studies', 'projects'],
+      invalidatesTags: (_result, _error, args) => [
+        { type: 'projects', id: 'LIST' },
```

Comment:
Why does this invalidate `{ type: 'projects', id: 'LIST' }`? This request doesn't create/delete a project right?